metals/src/main/scala/scala/meta/internal/metals/codeactions/PatternMatchRefactor.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Term
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class PatternMatchRefactor(trees: Trees) extends CodeAction {
+
+  override val kind: String = l.CodeActionKind.RefactorRewrite
+
+  private def convert(
+      patternMatch: Term.Match,
+      path: AbsolutePath
+  ): Seq[l.CodeAction] = {
+    patternMatch.expr match {
+      case _: Term.Placeholder =>
+        val range = patternMatch.pos.toLSP
+        val indentSize = range.getStart.getCharacter
+        // in order to replace current indent in text edit
+        range.getStart.setCharacter(0)
+
+        def stripMultilineCaseIndent(caze: meta.Case): String = {
+          val lines = caze.toString.split("\n")
+          (lines.take(1) ++ lines.drop(1).map(_.stripPrefix("  ")))
```

Comment:
We should calculate what indentation is being used instead of assuming double spaces (though this will be most widely used for sure.) We can check previous indentation before `_` and compare with the first after `_` and that difference could be used here.